wlauto/workloads/octane2/__init__.py

```diff
@@ -0,0 +1,183 @@
+    Octane 2 is a JavaScript benchmark for browser
+
+    """
+    octane_jar_file = 'org.linaro.wlauto.uiauto.octane2.jar'
+    octane_run_string = 'org.linaro.wlauto.uiauto.octane2.UiAutomation#runUiAutomation'
+
+    parameters = [
+        Parameter('force_dependency_push', kind=boolean, default=False,
+                  description=('Specifies whether to push dependency files to the device'
+                               'if they are already on it.')),
+        Parameter('perform_cleanup', kind=boolean, default=False,
+                  description='If ``True``, workload files on the device will be deleted after execution.'),
+        Parameter('browser_package', default='org.chromium.chrome',
+                  description='Specifies the package name of the device\'s browser app.'),
+        Parameter('browser_activity', default='.browser.ChromeTabbedActivity',
+                  description='Specifies the startup activity  name of the device\'s browser app.'),
+        Parameter('octane_archive', default="https://github.com/chromium/octane/archive/master.zip",
+                  description='URL of the octane zip file to be downloaded in case it is absent \
+                  from dependencies directory'),
+        Parameter('octane_filename', default="octane2.zip",
+                  description='File name that is downloaded from octane_archive'),
+        Parameter('octane_timeout', kind=int, default=900,
+                  description='Specifies the timeout (in seconds) before the uiautomator script is terminated.'),
+        Parameter('clear_file_cache', kind=boolean, default=True,
+                  description='Clear the the file cache on the target device prior to running the workload.'),
+    ]
+
+    supported_platforms = ['android']
+
+    def setup(self, context):  # NOQA
+        UiAutomatorWorkload.setup(self, context)
+        self.octane2_on_device = self.device.path.join(self.device.working_directory, self.name)
+        self.index_octane = 'file://{}'.format(self.octane2_on_device) + '/index.html'
+
+        # assume that it's enough to have index.html in the dependencies
+        if not os.path.isdir(self.dependencies_directory) or \
+              not os.path.exists(os.path.join(self.dependencies_directory, "index.html")):
+            self._download_octane2_file()
+
+        # Push the octane2
+        if self.force_dependency_push or not self.device.file_exists(
+                os.path.join(self.octane2_on_device, "index.html")):
+            self.logger.debug('Copying octane2 to device.')
+            self.device.push_file(self.dependencies_directory, self.octane2_on_device, timeout=300)
+
+        # Open the browser with default page
+        self.device.execute('am start -W -n  {}/{} about:blank'.format(self.browser_package, self.browser_activity))
+
+        # Stop the browser if already running and wait for it to stop
+        self.device.execute('am force-stop {}'.format(self.browser_package))
+
```

Comment:
This is copy/paste from bbench. Probably not necessary here but ensures the browser activity is loaded and unloaded before actual benchmark starts.